load-test-framework/src/main/proto/loadtest.proto

```diff
@@ -0,0 +1,75 @@
+syntax = "proto3";
+import "google/protobuf/duration.proto";
+import "google/protobuf/timestamp.proto";
+
+option java_package = "com.google.pubsub.flic.common";
+option java_outer_classname = "LoadtestProto";
+
+service Loadtest {
+  rpc Start(StartRequest) returns (StartResponse);
+  rpc Check(CheckRequest) returns (CheckResponse);
+}
+
+message StartRequest {
+  // The GCP project. This must be set even for Kafka, as we use it to export metrics to Stackdriver.
+  string project = 1;
+
+  // The Pub/Sub or Kafka topic name.
+  string topic = 2;
+
+  // The number of requests that can be made, each second, per client.
+  int32 request_rate = 3;
+
+  // The size of each user message to publish
+  int32 message_size = 4;
+
+  // The number of user messages of size message_size to batch together in each publish
+  int32 batch_size = 5;
+
+  // The maximum outstanding requests, per client.
+  int32 max_outstanding_requests = 6;
+
+  // The time at which the load test should start. If this is less than the current time, we start immediately.
+  google.protobuf.Timestamp start_time = 7;
+
+  oneof stop_conditions {
+    // The time at which the load test should be stopped. If less than the current time, the load test will not run.
```

Comment:
"If less than current time or less than start_time, the load test will not run."
I guess in this case, status will immediately be set to 'done'?